### PR TITLE
1.1.0

### LINKED ISSUE #25: rename "dont_catch_exceptions" parameter of ExperimentRunner.run method to "propogate_exceptions"
also update documentation of why this is useful (if you want to pause on exception while debugging)

### PR DESCRIPTION
**Pull request by octopuscabbage**
fixes #25 #23

### CODE REVIEW

**Comment by octopuscabbage** [comment]
I should probably go back and document stuff but it's mostly internal and I need this PR for the iserr project

**Comment by octopuscabbage** [comment]
This PR does technically break some backwards compatibility with 1.0.0 but I don't think it's worth bumping to 2.0.0 because of that.

File: smallab/checkpointed_experiment_handler.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+from smallab.specification_hashing import specification_hash
+from smallab.types import Specification
+
+
+class CheckpointedExperimentHandler():
+    """
+    An internal handler to handle running checkpointed experiments.
+    """
+    def __init__(self, rolled_backups=3):
+        self.rolled_backups = rolled_backups
```
Comment (akumaraguru) [inline smallab/checkpointed_experiment_handler.py:21]: Maybe add a docstring describing what `rolled_backups` does?

File: smallab/experiment.py [hunk 2]
```diff
@@ -53,3 +59,29 @@ def main(self, specification: Specification) -> typing.Dict:
         :return: A dictionary of results which will be serialized and saved if this experiment is succesful
         """
         pass
+
+
+class CheckpointedExperiment(BaseExperiment):
+    """
+    CheckpointedExperiment is an Experiment which can be stopped and restarted.
+    Step is called multiple times, after it returns the current experiment is serialized allowing the experiment to
+    restart seemlessly since to step it will not look any different whether python has shut down in between succesive calls.
```
[the remark this answers is not in the document]
Comment (akumaraguru) [reply smallab/experiment.py:68]: You should also mention somewhere, if not here, that checkpointing will fail if the experiment contains objects that are not picklable.

File: smallab/checkpointed_experiment_handler.py (hunk 1 above)
Comment (octopuscabbage) [reply smallab/checkpointed_experiment_handler.py:21]: as of right now there's no way of an end user configuring this, but i'll add docs just in case

**Pull request merged**